## Report No.01 自力でやってみよう

1. sin(x)=3/4の近似解を求めてみよう。

In [1]:
g(x) = sin(x) -3/4      #関数を定義
dg(x) = cos(x)          #微分した関数も定義

x = 1.0       #初期値を設定
sq = [x]

for i in 1:100
    println(i, ": ", x, ", ", g(x)) 
    if abs(g(x)) < 1.0e-06          
        break
    end
    x = x - g(x)/dg(x)  
    push!(sq, x)        # 新しい近似値をベクトルの要素として増やす
end

1: 1.0, 0.0914709848078965
2: 0.830704063605792, -0.011593656366530647
3: 0.847896252361635, -0.00010969431063945212
4: 0.8480620633957656, -1.0308981890361224e-8


よってsin(x)=3/4の近似解はx=0.8480620633957656と求まる。

---

2.連立方程式
\begin{cases}
 {}
 x + 2y = -1, & \\
 sin(x+y) = 0.7 &
\end{cases}
の近似解を求めてみよう。

In [3]:
function g(vx)
    x,y = vx
    return [
        x+2y+1
        sin(x+y)-0.7
    ]
end

function J(vx)
        x,y = vx
        return [
            1 2
            cos(x+y) cos(x+y)
        ]
end

vx = [ 1.0, 0.0 ]   #初期値の設定

sq = copy(vx)

for i in 1:100
    println(i, ": ", vx,", ", g(vx))
    if norm(g(vx)) < 1.0e-06
        break
    end
    vx = vx - J(vx)\g(vx)
    sq = hcat(sq, vx)
end

1: [1.0, 0.0], [2.0, 0.141471]
2: [2.47633, -1.73816], [0.0, -0.0270696]
3: [2.54952, -1.77476], [0.0, -0.000456585]
4: [2.55079, -1.7754], [0.0, -1.42827e-7]


以上が連立方程式の解である。